lib/errors.js

```diff
@@ -15,7 +16,7 @@ function error(code, data, err) {
     if (data) {
         // ** If the second argument is an error, then reset the arguments.
         if (isError(data))
-            return error(code, null, data);
+            return error(code, null, data, err);
```

Comment:
if  data is the error, you would recive the stack, if i understood correcly, this is the way to use it as overload in js